SpatialConnect/SCExchangeBackend.m

```diff
@@ -0,0 +1,192 @@
+      [configService setCachedConfig:cfg];
+      [configService loadConfig:cfg];
+
+      SpatialConnect *sc = [SpatialConnect sharedInstance];
+      [sc.backendService.configReceived sendNext:@(YES)];
+    }
+  }];
+}
+
+- (void)loadCachedConfig {
+  SCConfig *config = [configService cachedConfig];
+  SpatialConnect *sc = [SpatialConnect sharedInstance];
+
+  if (config) {
+    [configService loadConfig:config];
+    [sc.backendService.configReceived sendNext:@(YES)];
+  }
+}
+
+- (void)listenForSyncEvents {
+}
+
+- (NSMutableArray *)fetchLayerNames {
+  NSMutableArray *layerNames = [NSMutableArray new];
+  NSURL *url =
+      [NSURL URLWithString:[NSString stringWithFormat:@"%@/gs/acls",
+                                                      remoteConfig.httpUri]];
+
+  NSString *authHeader =
+      [NSString stringWithFormat:@"Bearer %@", [authService xAccessToken]];
+  NSDictionary *res =
+      [SCHttpUtils getRequestURLAsDictBLOCKING:url auth:authHeader];
+
+  NSMutableArray *rw = res[@"rw"];
+  [rw enumerateObjectsUsingBlock:^(NSString *l, NSUInteger idx, BOOL *stop) {
+    [layerNames addObject:l];
+  }];
+
+  return layerNames;
+}
+
+- (SCFormConfig *)buildSCFormConfig:(NSString *)layerName {
+
+  NSURL *url =
+      [NSURL URLWithString:[NSString stringWithFormat:@"%@/layers/%@/get",
+                                                      remoteConfig.httpUri,
+                                                      layerName]];
+  NSString *authHeader =
+      [NSString stringWithFormat:@"Bearer %@", [authService xAccessToken]];
+
+  NSDictionary *res =
+      [SCHttpUtils getRequestURLAsDictBLOCKING:url auth:authHeader];
+
+  NSMutableArray *attributes = res[@"attributes"];
+  NSMutableDictionary *d = [[NSMutableDictionary alloc] init];
+  [d setObject:layerName forKey:@"form_key"];
+  [d setObject:res[@"title"] forKey:@"form_label"];
+  [d setObject:[NSNumber numberWithInteger:1] forKey:@"version"];
+  [d setObject:[NSNumber numberWithInteger:[self randomNumberBetween:1
+                                                           maxNumber:100000]]
+         forKey:@"id"];
+
+  NSMutableArray *fields = [NSMutableArray new];
+  [attributes enumerateObjectsUsingBlock:^(NSDictionary *attribute,
+                                           NSUInteger idx, BOOL *stop) {
+    NSDictionary *fieldDict = @{
+      @"field_key" : attribute[@"attribute"],
+      @"field_label" : attribute[@"attribute"],
+      @"type" : [self fieldType:attribute],
+      @"position" : attribute[@"display_order"]
+    };
+    [fields addObject:fieldDict];
+  }];
+  [d setObject:fields forKey:@"fields"];
+
+  return [[SCFormConfig alloc] initWithDict:d];
+  ;
+}
+
+- (NSInteger)randomNumberBetween:(NSInteger)min maxNumber:(NSInteger)max {
+  return min + arc4random_uniform((uint32_t)(max - min + 1));
+}
+
+- (NSString *)fieldType:(NSDictionary *)attribute {
+
+  if ([attribute[@"attribute"] isEqualToString:@"photo"]) {
```

Comment:
This could probably be a dictionary lookup and would be easier to read
```
dict = @{@"xsd:int":@"number",
     @"psd:float":@"number",
     ...
}
[dict stringForKey:t];
```